### PR TITLE
Optional vpc

### LINKED ISSUE #3: Make VPC optional for aws deployment

### PR DESCRIPTION
**Pull request by mickmcgrath13**
closes #3

### CODE REVIEW

**Comment by mickmcgrath13** [comment]
Tested via internal bitovi ops repo (operations-stackstorm)

File: action.yaml [hunk 1]
```diff
@@ -46,6 +46,10 @@ inputs:
   # create_hosted_zone:
   #   description: "Bool, whether the hosted zone should be created in the action. Default is false"
   #   default: false
+  create_vpc:
+    description: "Bool, whether a VPC should be created in the action. Default is false"
+    default: false
```
Comment (armab) [inline action.yaml:51]: This is helpful and makes a lot of sense :+1:
In prod users would probably manage VPC themselves, custom created or community module like https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws/latest.

Looks like, however, the PR doesn't provide enough configuration to re-use the existing VPC or subnet specified by the user.

File: operations/deployment/terraform/modules/02_networking.tf [hunk 2]
```diff
@@ -104,7 +103,7 @@ resource "aws_security_group" "allow_https" {
 resource "aws_security_group" "allow_ssh" {
  name        = "allow_ssh"
  description = "Allow SSH traffic"
- vpc_id      = aws_vpc.main.id
+ vpc_id      = var.create_vpc == "true" ? aws_vpc.main[0].id : null
```
Comment (armab) [inline operations/deployment/terraform/modules/02_networking.tf:106]: If the user is managing VPC outside of this GH Action, we'll need to support specifying the custom VPC ID for the security group and other resources.

File: operations/deployment/terraform/modules/02_networking.tf [hunk 3]
```diff
@@ -85,7 +84,7 @@ resource "aws_security_group" "allow_http" {
 resource "aws_security_group" "allow_https" {
  name        = "allow_https"
```
Comment (armab) [inline operations/deployment/terraform/modules/02_networking.tf:85]: BTW the naming is too generic to be re-used in some existing VPC that might lead to conflicts.
Something like `st2_allow_https` might work better

File: operations/deployment/terraform/modules/02_networking.tf (hunk 2 above)
Comment (mickmcgrath13) [reply operations/deployment/terraform/modules/02_networking.tf:106]: yep! https://github.com/bitovi/github-actions-deploy-stackstorm/issues/4

want to add a readme and publish to GHA marketplace. @LeoDiazL is working on the `vpc_id` option

File: action.yaml (hunk 1 above)
Comment (mickmcgrath13) [reply action.yaml:51]: yep! https://github.com/bitovi/github-actions-deploy-stackstorm/issues/4

want to add a readme and publish to GHA marketplace. @LeoDiazL is working on the `vpc_id` option

File: operations/deployment/terraform/modules/02_networking.tf (hunk 2 above)
Comment (armab) [reply operations/deployment/terraform/modules/02_networking.tf:106]: 👍

File: operations/deployment/terraform/modules/02_networking.tf (hunk 3 above)
Comment (mickmcgrath13) [reply operations/deployment/terraform/modules/02_networking.tf:85]: captured here: https://github.com/bitovi/github-actions-deploy-stackstorm/issues/8

**Pull request merged**